### PR TITLE
Article: Strings in Ruby

### PR DESCRIPTION
**Pull request by gwenf**
Created an article called Ruby-Strings.md

### CODE REVIEW

**Comment by iheartkode** [comment]
Looks Good, thanks for the awesome Ruby post :+1:

File: Ruby-Strings.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Ruby Strings
+
+**Basics:**
+
+ * Strings are a series of characters 'strung' together between quotes.
+ * Single or double quotes can be used to create strings in Ruby.
+ * Ruby does some extra evaluation on strings that are created with double quotes, such as:
+ 	*	Escaping characters: \n, \t, \s
```
Comment (alayek) [inline Ruby-Strings.md]: > Escaping characters: ~~\n~~ `\n`, ~~\t~~ `\t`, ~~\s~~ `\s`

File: Ruby-Strings.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+ 	*	Using variables and expressions inside: #{variable or expression}
```
Comment (alayek) [inline Ruby-Strings.md]: > Using variables and expressions inside: ~~#{variable or expression}~~ `#{variable or expression}`

File: Ruby-Strings.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+ * Strings with single quotes are rendered as they are without any special considerations.
+
+**Examples:**
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]:
```diff
- ```
+ ```ruby
```

File: Ruby-Strings.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+"Hello World"
+# is equivilant to:
+'Hello World'
+```
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+"This is line 1.\nAnd this is line 2."
+# returns:
+This is line 1.
+And this is line 2.
+```
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+name = "Batman"
+"Hello, my name is #{name}!"
+# returns:
+Hello, my name is Batman!
+```
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+
+## String Operations:
+
+**Concatenation:**
+
+*	Strings can be joined together using the '+' operator.
+
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+# returns:
+This is your name:\n#{name}
+```
+
+
+## String Operations:
+
+**Concatenation:**
+
+*	Strings can be joined together using the '+' operator.
```
Comment (alayek) [inline Ruby-Strings.md]: > Strings can be joined together using the ~~'+'~~ `+` operator.

File: Ruby-Strings.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+
+```
+"Hello" + " World" + "!"
+# returns:
+Hello World!
+```
+
+**Multiplication:**
+
+*	Strings can be multiplied by an integer value using the '*' operator.
```
Comment (alayek) [inline Ruby-Strings.md]: > Strings can be multiplied by an integer value using the ~~'*'~~ `*` operator.

File: Ruby-Strings.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+HelloHelloHello
+```
+
+
+
+## String Methods:
+
+### length:
+
+*	The .length property returns the number of characters in a string including white-space.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.length~~ `.length` property returns

File: Ruby-Strings.md [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+"Hello".length
+# returns: 5
+```
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+"Hello World!".length
+# returns: 12
+```
+
+### count:
+
+*	The .count method counts how many times a specific character(s) is found in a string.
+*	This method is case-sensitive.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 15]
```diff
@@ -0,0 +1,194 @@
...
+"HELLO".count('L')  
+# returns: 2
+```
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+"HELLO WORLD!".count('LO')  
+# returns: 1
+```
+
+### insert:
+
+*	The .insert method inserts a string into another string before a given index.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+# returns:
+Helhi5lo
+# "hi5" is inserted into the string right before the second 'l' which is at index 3
+```
+
+### upcase:
+
+*	The .upcase method transforms all letters in a string to uppercase.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+"Hello".upcase
+# returns:
+HELLO
+```
+
+### downcase:
+
+*	The .downcase method transforms all letters in a string to lowercase.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+"Hello".downcase
+# returns:
+hello
+```
+
+### capitilize:
+
+*	The .capitalize method make the first letter in a string uppercase and the rest of the string lowercase.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 20]
```diff
@@ -0,0 +1,194 @@
...
+"HELLO".capitalize
+# returns:
+Hello
+```
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 21]
```diff
@@ -0,0 +1,194 @@
...
+"HELLO, HOW ARE YOU?".capitalize
+# returns:
+Hello, how are you?
+```
+*Note that the first letter is only capitilized if it is at the beginning of the string:*
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 22]
```diff
@@ -0,0 +1,194 @@
...
+# returns:
+-hello
+1hello
+```
+
+### reverse:
+
+*	The .reverse method reverses the order of the characters in a string.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 23]
```diff
@@ -0,0 +1,194 @@
...
+# returns:
+"!dlroW olleH"
+```
+
+### split:
+
+*	The .split takes a strings and 'splits' it into an array.
+*	The default method splits the string based on whitespace.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 24]
```diff
@@ -0,0 +1,194 @@
...
+"Hello, how are you?".split
+# returns:
+["Hello,", "how", "are", "you?"]
+```
+
+### chop:
+
+*	The .chop method removes the last character of the string.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 25]
```diff
@@ -0,0 +1,194 @@
...
+"Name".chop
+# returns:
+Nam
+```
+
+### strip:
+
+*	The .strip method removes the leading and trailing whitespace on strings.
+
+```
```
Comment (alayek) [inline Ruby-Strings.md]: Same as [above](#r62454004)

File: Ruby-Strings.md [hunk 26]
```diff
@@ -0,0 +1,194 @@
...
+# returns: 5
+```
+```
+"Hello World!".length
+# returns: 12
+```
+
+### count:
+
+*	The .count method counts how many times a specific character(s) is found in a string.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.count~~ `.count` method counts

File: Ruby-Strings.md [hunk 27]
```diff
@@ -0,0 +1,194 @@
...
+# returns: 2
+```
+```
+"HELLO WORLD!".count('LO')  
+# returns: 1
+```
+
+### insert:
+
+*	The .insert method inserts a string into another string before a given index.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.insert~~ `.insert` method inserts

File: Ruby-Strings.md [hunk 28]
```diff
@@ -0,0 +1,194 @@
...
+```
+"Hello".insert(3, "hi5")
+# returns:
+Helhi5lo
+# "hi5" is inserted into the string right before the second 'l' which is at index 3
+```
+
+### upcase:
+
+*	The .upcase method transforms all letters in a string to uppercase.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.upcase~~ `.upcase` method transforms

File: Ruby-Strings.md [hunk 29]
```diff
@@ -0,0 +1,194 @@
...
+
+```
+"Hello".upcase
+# returns:
+HELLO
+```
+
+### downcase:
+
+*	The .downcase method transforms all letters in a string to lowercase.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.downcase~~ `.downcase` method transforms

File: Ruby-Strings.md [hunk 30]
```diff
@@ -0,0 +1,194 @@
...
+
+```
+"Hello".downcase
+# returns:
+hello
+```
+
+### capitilize:
+
+*	The .capitalize method make the first letter in a string uppercase and the rest of the string lowercase.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.capitalize~~ `.capitalize` method make the

File: Ruby-Strings.md [hunk 31]
```diff
@@ -0,0 +1,194 @@
...
+"HELLO".capitalize
+# returns:
+Hello
+```
+```
+"HELLO, HOW ARE YOU?".capitalize
+# returns:
+Hello, how are you?
+```
+*Note that the first letter is only capitilized if it is at the beginning of the string:*
```
Comment (alayek) [inline Ruby-Strings.md]: > beginning of the string:

Remove the colon at the end.

File: Ruby-Strings.md [hunk 32]
```diff
@@ -0,0 +1,194 @@
...
+
+```
+"Hello World!".reverse
+# returns:
+"!dlroW olleH"
+```
+
+### split:
+
+*	The .split takes a strings and 'splits' it into an array.
```
Comment (alayek) [inline Ruby-Strings.md]:
```diff
- The .split takes a strings and 'splits' it into an array.
+ The `split` takes a string and *splits* it into an array, then returns the array
```

File: Ruby-Strings.md [hunk 33]
```diff
@@ -0,0 +1,194 @@
...
+*	The default method splits the string based on whitespace.
```
Comment (alayek) [inline Ruby-Strings.md]: > The default method splits the string based on whitespace.

Add what you mean by _default method_. I am guessing you mean that the separator is not explicitly provided?

File: Ruby-Strings.md [hunk 34]
```diff
@@ -0,0 +1,194 @@
...
+
+```
+"Hello, how are you?".split
+# returns:
+["Hello,", "how", "are", "you?"]
+```
+
+### chop:
+
+*	The .chop method removes the last character of the string.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.chop~~ `.chop` method removes the last character of the string.

I have a question here. Does it mutate the original String, or does it return a new String?

File: Ruby-Strings.md [hunk 35]
```diff
@@ -0,0 +1,194 @@
...
+
+```
+"Name".chop
+# returns:
+Nam
+```
+
+### strip:
+
+*	The .strip method removes the leading and trailing whitespace on strings.
```
Comment (alayek) [inline Ruby-Strings.md]: > The ~~.strip~~ `.strip` method removes the leading and trailing whitespace on strings.

Does it also remove newlines, CRLF etc.?

**Comment by alayek** [comment]
@gwenf thanks for the article. Awesome work! Please take care of the comments I have left, and we are great to go :+1: 

@theflametrooper @koustuvsinha might wanna take a look.

File: Ruby-Strings.md [hunk 36]
```diff
@@ -0,0 +1,194 @@
+# Ruby Strings
+
+**Basics:**
+
+ * Strings are a series of characters 'strung' together between quotes.
+ * Single or double quotes can be used to create strings in Ruby.
+ * Ruby does some extra evaluation on strings that are created with double quotes, such as:
+ 	*	Escaping characters: \n, \t, \s
+ 	*	Using variables and expressions inside: #{variable or expression}
+ * Strings with single quotes are rendered as they are without any special considerations.
```
Comment (theflametrooper) [inline Ruby-Strings.md]: ~~as they are without~~ as they are, without

File: Ruby-Strings.md [hunk 37]
```diff
@@ -0,0 +1,194 @@
...
+
+**Examples:**
+
+```
+"Hello World"
+# is equivilant to:
```
Comment (theflametrooper) [inline Ruby-Strings.md]: ~~equivilant~~ equivalent

File: Ruby-Strings.md [hunk 38]
```diff
@@ -0,0 +1,194 @@
...
+'Hello World'
+```
+```
+"This is line 1.\nAnd this is line 2."
```
Comment (theflametrooper) [inline Ruby-Strings.md]: ~~\nAnd this~~ `\n` And this

File: Ruby-Strings.md (hunk 34 above)
Comment (gwenf) [inline Ruby-Strings.md]: It returns a new string unless you use .chop! which mutates the original

File: Ruby-Strings.md (hunk 33 above)
Comment (gwenf) [inline Ruby-Strings.md]: Yes, that's what I meant - if no arguments are provided...

**Comment by Rafase282** [comment]
I would separate into multiple articles, but other than that it seems fine.

**Comment by theflametrooper** [comment]
@Rafase282 I think it's fine, it's not that long of an article.

**Comment by alayek** [comment]
@Rafase282 I was gonna point that out in one of the comments, and in fact this article would get bigger once we actively start Ruby curriculum development. I think then we can split.

**Comment by Rafase282** [comment]
I agree, but we split on this or a new PR though.
Are you up for it @gwenf

**Comment by koustuvsinha** [comment]
# LGTM :+1:

**Comment by gwenf** [comment]
@theflametrooper @alayek I am working on splitting the article now. Shouldn't take too long.

**Comment by theflametrooper** [comment]
@gwenf Awesome!

**Comment by theflametrooper** [comment]
@gwenf Any update on the splitting? :P

File: Ruby-String-Methods.md [hunk 39]
```diff
@@ -0,0 +1,146 @@
+
+## String Methods:
```
Comment (Rafase282) [inline Ruby-String-Methods.md]:
```diff
-## String Methods:
+# String Methods
```

File: Ruby-String-Methods.md [hunk 40]
```diff
@@ -0,0 +1,146 @@
...
+
+### length:
```
Comment (Rafase282) [inline Ruby-String-Methods.md]:
```diff
-### length:
+## Length:
```

**Comment by Rafase282** [comment]
Great job, as I pointed out in one of the files. Please use `h1` for the title, no `:` for that one. The rest should be `h2`.

File: Ruby-String-Methods.md [hunk 41]
```diff
@@ -0,0 +1,146 @@
+
```
Comment (Rafase282) [inline Ruby-String-Methods.md]: Please remove this empty line.

File: Ruby-String-Methods.md [hunk 42]
```diff
@@ -0,0 +1,146 @@
...
+# String Methods:
+
+## length:
```
Comment (Rafase282) [inline Ruby-String-Methods.md]: Given that this is a heading, should the first letter not be capitalize?

File: Ruby-String-Operators.md [hunk 43]
```diff
@@ -0,0 +1,29 @@
+# String Operations
+
+**Concatenation:**
```
Comment (Rafase282) [inline Ruby-String-Operators.md]: Please change these to `h2` headings.

File: Ruby-String-Operators.md [hunk 44]
```diff
@@ -0,0 +1,29 @@
...
+"Hello" * 3
+# returns:
+HelloHelloHello
+```
+
+---
+
+| Previous | Home | Next |
+| :---         |     :---:      |          ---: |
+| [Ruby Basics](Ruby-Strings.md)   | [Table of Contents](Ruby-Strings.md)      | [Ruby String Methods](Ruby-String-Methods.md)    |
```
Comment (Rafase282) [inline Ruby-String-Operators.md]: There is no need to add the `.md`

File: Ruby-String-Methods.md [hunk 45]
```diff
@@ -0,0 +1,145 @@
...
+```
+```ruby
+name = "Batman"
+name.chop!
+name == "Batma" # returns true
+```
+
+## Strip:
+
+*	The ```.strip``` method removes the leading and trailing whitespace on strings, including tabs, newlines, and carriage returns (```\t```, ```\n```, ```\r```).
```
Comment (Rafase282) [inline Ruby-String-Methods.md]: For inline commenting you can use a single tick instead of three.

File: Ruby-Strings.md [hunk 46]
```diff
@@ -0,0 +1,46 @@
...
+**Basics:**
+
+ * Strings are a series of characters 'strung' together between quotes.
+ * Single or double quotes can be used to create strings in Ruby.
+ * Ruby does some extra evaluation on strings that are created with double quotes, such as:
+ 	*	Escaping characters: `\n`, `\t`, `\s`
+ 	*	Using variables and expressions inside: `#{variable or expression}`
+ * Strings with single quotes are rendered as they are, without any special considerations.
+
+**Examples:**
```
Comment (Rafase282) [inline Ruby-Strings.md]: Please turn this into a `h2` to avoid displaying too much on gitter chat.

Comment (gwenf) [inline Ruby-Strings.md]: Ok, I did it

**Comment by Rafase282** [comment]
# LGTM :+1:

File: Ruby-String-Methods.md [hunk 47]
```diff
@@ -0,0 +1,145 @@
+# String Methods
+
```
Comment (Rafase282) [inline Ruby-String-Methods.md]: Please add a brief introduction about the string method.

File: Ruby-String-Operators.md [hunk 48]
```diff
@@ -0,0 +1,29 @@
+# String Operations
+
```
Comment (Rafase282) [inline Ruby-String-Operators.md]: Please add a brief introduction about the string operations.

**Comment by alayek** [comment]
# LGTM :+1:

**Comment by theflametrooper** [comment]
#### :sparkles: :sparkles: **LGTM :+1:** :sparkles: :sparkles: 

:tada: Thanks for contributing @gwenf  :tada:

**Pull request merged**